torchx/CMakeLists.txt

```diff
@@ -33,6 +34,10 @@ else()
     # Although the compiler complains about not using these,
     # things only work with them set
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -undefined dynamic_lookup")
+    check_c_compiler_flag("-arch arm64" ARM64_SUPPORTED)
+    if(ARM64_SUPPORTED)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DMAC_ARM64")
```

Comment:
I've tested this both on Mac M1 and Mac Intel and it's working as expected. The CI tests windows and linux accordingly